src/foam/nanos/rope/test/TestObjects.js

```diff
@@ -67,63 +67,66 @@ foam.CLASS({
   ]
 });
 
+foam.RELATIONSHIP({
+  cardinality: '*:*',
+  sourceModel: 'foam.nanos.rope.test.ROPEUser',
+  targetModel: 'foam.nanos.rope.test.ROPEUser',
+  forwardName: 'entities',
+  inverseName: 'agents',
+  junctionDAOKey: 'ropeAgentJunctionDAO'
+});
+
+foam.RELATIONSHIP({
+  cardinality: '*:*',
+  sourceModel: 'foam.nanos.rope.test.ROPEUser',
+  targetModel: 'foam.nanos.rope.test.ROPEUser',
+  forwardName: 'parteners',
+  inverseName: 'partnered',
+  junctionDAOKey: 'ropePartnerJunctionDAO'
+});
+
+foam.RELATIONSHIP({
+  cardinality: '1:*',
+  sourceModel: 'foam.nanos.rope.test.ROPEUser',
+  targetModel: 'foam.nanos.rope.test.ROPEUser',
+  forwardName: 'contacts',
+  inverseName: 'owner',
+  junctionDAOKey: 'ropeContactDAO'
+});
+
+foam.RELATIONSHIP({
+  cardinality: '*:*',
+  sourceModel: 'foam.nanos.rope.test.ROPEBusiness',
+  targetModel: 'foam.nanos.rope.test.ROPEUser',
+  forwardName: 'signingOfficers',
+  inverseName: 'businessesInWhichThisUserIsASigningOfficer',
+  junctionDAOKey: 'ropeSigningOfficerJunctionDAO'
+});
+
+foam.RELATIONSHIP({
+  cardinality: '1:*',
+  sourceModel: 'foam.nanos.rope.test.ROPEUser',
+  targetModel: 'foam.nanos.rope.test.ROPEBankAccount',
+  forwardName: 'bankaccounts',
+  inverseName: 'owner',
+  junctionDAOKey: 'ropeAccountDAO'
+});
+
+foam.RELATIONSHIP({
+  cardinality: '1:*',
+  sourceModel: 'foam.nanos.rope.test.ROPEBankAccount',
+  targetModel: 'foam.nanos.rope.test.ROPETransaction',
+  forwardName: 'debits',
+  inverseName: 'sourceAccount',
+  junctionDAOKey: 'ropeTransactionDAO'
+});
+
+foam.RELATIONSHIP({
+  cardinality: '1:*',
+  sourceModel: 'foam.nanos.rope.test.ROPEBankAccount',
+  targetModel: 'foam.nanos.rope.test.ROPETransaction',
+  forwardName: 'credits',
+  inverseName: 'sourceAccount',
+  junctionDAOKey: 'ropeTransactionDAO'
+});
```

Comment:
plz add the junction models generated from the *:* relationships to foam2/tools/classes.js under the ROPE section